api/src/pcapi/core/mails/__init__.py

```diff
@@ -42,7 +42,7 @@ def get_contact_url(contact_email: str) -> str | None:
 def _get_backend(data: models.TransactionalEmailData | models.TransactionalWithoutTemplateEmailData) -> type:
     # Do not send unnecessary transactional emails through Sendinblue in EHP
     if (
-        (settings.IS_STAGING or settings.IS_TESTING)
+        settings.DO_NOT_SEND_TRANSACTIONNAL_EMAIL
```

Comment:
Le nommage est trompeur : même avec `DO_NOT_SEND_TRANSACTIONNAL_EMAIL` activé, on envoie bien les mails pour lesquels le template a `send_to_ehp == True`.

Je n'ai pas de meilleur nommage... à part définir un `IS_EHP = not (IS_PROD or IS_INTEGRATION)`, mais ça me paraît un peu gratuit...